src/internal/channel.js

```diff
@@ -113,7 +113,7 @@ export function channel(buffer) {
 }
 
 export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
-  if(arguments.length > 1)
+  if(typeof matcher !== 'undefined')
```

Comment:
As a side note initially the buffer argument was checked using arguments.length like match, but since the arg is used internally by the lib (to resolve the `yield channel` effects) I adopted the default value approach because it makes the internal client code less complicated